packages/builders/scripts/vats/upgrade-paRegistry.js

```diff
@@ -0,0 +1,21 @@
+import { makeHelpers } from '@agoric/deploy-script-support';
+
+/** @type {import('@agoric/deploy-script-support/src/externalTypes.js').CoreEvalBuilder} */
+export const defaultProposalBuilder = async ({ publishRef, install }) =>
+  harden({
+    sourceSpec: '@agoric/vats/src/proposals/upgrade-paRegistry-proposal.js',
+    getManifestCall: [
+      'getManifestForUpgradingRegistry',
+      {
+        registryRef: publishRef(
+          install('@agoric/vats/src/vat-priceAuthority.js'),
```

Comment:
Will this update `agoricNames.installation` with a new zoe `Installation`? Since this isn't a contract, that would be odd.

Reply:
Oh! good point. I'll have to figure out how to pass the bundle through without installing a contract.

 We use `publishRef(install())` in the Zoe proposals.  I don't know why that's okay.

Reply:
> We use publishRef(install()) in the Zoe proposals. I don't know why that's okay, but it seems to be the standard approach.

It was certainly the standard approach for the initial bootstrap of the system. Whether the idiom is appropriate for upgrades, I am not at all sure. @michaelfig , thoughts?

Reply:
`publishRef(install(...))` is needed to create a bundlecap before running the `getManifest*` behaviours.  This is appropriate for all circumstances when a contract needs to be taken from disk and its bundlecap passed in to the `getManifest*` call.

It's the `getManifestForUpgradingRegistry` that controls how those bundlecaps are passed to the behaviours.  See below...

Reply:
> This is appropriate for all circumstances when a contract needs to be taken from disk and its bundlecap passed in to the getManifest* call.

@michaelfig:  The issue @dckc is bringing up applies for non-contract vats. Is there a way to produce a bundleCap that can be used for pure vats that doesn't presume to install it as if it were a Zoe contract?

Reply:
In `const bundleRef = publishRef(install('entrypoint.js'))`, the `install` creates an on-chain bundleCap (*not* a Zoe installation) from `entrypoint.js`, and `publishRef` produces a `VatSourceRef` for it, all with no inherent dependency on Zoe.

The only time Zoe installations would be implicitly created and published under `agoricNames.installation` would be if your `getManifest` return value had an `installations` property, which it doesn't.  Even if it did, the bundleRef still describes the underlying generic bundleCap, which has nothing to do with Zoe.

Other than that, you can put on your ocap reasoning cap to understand that the behaviours named by your `manifest` do not reference `zoe`, so they can't use Zoe.  Furthermore, your behaviours correctly use the vat's `adminNode` to upgrade the non-contract vat to the bundleCap corresponding to the bundleRef.

Reply:
> This is appropriate for all circumstances when a contract needs to be taken from disk...

Sorry for the confusion; I mistyped.  It should have been "... all circumstances when an [arbitrary JS module and its dependencies need] to be taken from disk...".